apps/jsonrpc2/lib/jsonrpc2/servers/websocket_http.ex

```diff
@@ -20,26 +228,43 @@ defmodule JSONRPC2.Servers.WebSocketHTTP do
   which are allowed in `cowboy_opts`.
 
   If the server `ref` is not set in `cowboy_opts`, `handler.HTTP` or `handler.HTTPS` is the default.
-  """
-  @spec child_spec(:http | :https, :web_ws | :web | :ws, module, list) :: Supervisor.child_spec()
-  def child_spec(scheme, type, handler, cowboy_opts \\ []) do
-    cowboy_opts = Keyword.merge(cowboy_opts, ref: ref(scheme, JSONRPC2Plug))
-    dispatch = get_dispatcher(type, handler)
-    do_child_spec(scheme, handler, Keyword.merge(cowboy_opts, dispatch))
+  '''
+
+  @spec child_spec(:web_ws | :web | :ws, module, configuration) :: Supervisor.child_spec()
+  defp child_spec(type, handler, configuration) do
+    port = Map.get(configuration, :port)
+
+    ip =
+      case Map.get(configuration, :interface) do
+        :local -> {0, 0, 0, 0}
+        :all -> :any
+        ip -> ip
```

Comment:
Are you validating provided ip anywhere?

Reply:
No. I decided that input validations should be validated beforehand on the CLI argument parsing.